datalad/interface/results.py

```diff
@@ -227,13 +228,18 @@ def annexjson2result(d, ds, **kwargs):
         res['metadata'] = {k: v[0] if isinstance(v, list) and len(v) == 1 else v
                            for k, v in d['fields'].items()
                            if not k.endswith('lastchanged')}
-    # avoid meaningless standard messages
-    if 'note' in d:
+    if d.get('error-messages', None):
+        messages.extend(d['error-messages'])
+    # avoid meaningless standard messages, and collision with actual error
+    # messages
+    elif 'note' in d:
```

Comment:
I do agree that errors are more important, but I wonder if `note` should also be retained? 
i.e. `elif` to be changed back to `if`?

Reply:
That is the example I posted. I think it makes it worse.

Reply:
and there is no way to keep error message "separate" so it could e.g. get a distinctive color (red) while rendering the result?

Reply:
We don't have that yet, but I recorded the need https://github.com/datalad/datalad/issues/3939